src/main/java/mate/academy/dao/BookDaoImpl.java

```
@@ -0,0 +1,127 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = parseBook(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = parseBook(resultSet);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Database error while finding all books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows > 0) {
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update book", e);
+        }
+        return null;
```

`update()` returns `null` when no rows are affected (line 110). The interface allows returning `Book`, but returning null can hide failures. Consider changing behavior to either throw a `DataProcessingException("Can't update book " + book, null)` if you treat it as an error, or clearly document the null return and handle it in callers. The checklist suggests verifying affected rows and reacting appropriately (see DAO guidance) .